user/user-tool-prefs/tool/src/java/org/sakaiproject/user/tool/UserPrefsTool.java

```diff
@@ -2286,6 +2304,9 @@ public String processHiddenSites()
 			hiddenUpdated = true;
 
 			m_reloadTop = Boolean.TRUE;
+			
+			eventTrackingService.post(eventTrackingService.newEvent(PREFS_HIDE_SITE_EVENT, getUserId(), true));
```

Comment:
BasePreferencesService commit already posts an event, but looks like it might not have an event set on it. I'd think this would be better done either calling setEvent on the edit to have an event set by the final commit. Couldn't we just listen for the SECURE_EDIT_PREFS (prefs.upd) one and clear that? That seems simpler overall.

```
			// track it
			eventTrackingService()
					.post(eventTrackingService().newEvent(((BasePreferences) edit).getEvent(), edit.getReference(), true));

			// close the edit object
			((BasePreferences) edit).closeEdit();
		}
```

Reply:
I'd remove this custom event from the tool, call a 
m_edit.setEvent(PreferencesService.SECURE_EDIT_PREFS) //If not set

Then do a listen for that in BaseSiteService. It feels like any change of preferences might want a user cache refreshed.